lib/src/main/kotlin/com/swmansion/starknet/data/types/Felt.kt

```diff
@@ -51,8 +76,45 @@ data class Felt(val value: BigInteger) : Comparable<Felt> {
         @field:JvmField
         val ONE = Felt(BigInteger.ONE)
 
+        /**
+         * Create Felt from hex string. It must start with "0x" prefix.
+         *
+         * @param value hex string.
+         */
         @JvmStatic
         fun fromHex(value: String): Felt = Felt(parseHex(value))
+
+        /**
+         * Create Felt from ASCII string. It must be shorter than 32 characters and only contain ASCII encoding.
+         * Example: "hello" -> 0x68656c6c6f.
+         *
+         * @param value string transformed to felt.
+         */
+        @JvmStatic
+        fun fromShortString(value: String): Felt {
+            if (value.length > 31) {
+                throw IllegalArgumentException("Short string cannot be longer than 31 characters.")
+            }
+            if (!isAscii(value)) {
+                throw IllegalArgumentException("String to be encoded must be an ascii string.")
+            }
+
+            val encoded = value.replace(Regex(".")) { s ->
+                s.value.first().code.toString(16).padStart(2, '0')
+            }
+
+            return fromHex("0x$encoded")
+        }
+
+        private fun isAscii(string: String): Boolean {
+            for (char in string) {
+                if (char.code < 0 || char.code > 127) {
```

Comment:
should control characters (0-31) be included?

Reply:
I think they should, they are included in python at least (`text.encode("ascii")` is used).